### ISSUE TITLE
Ebola vignette

### ISSUE THREAD

**Issue by athowes** [epinowcast/epidist#48]
## Goal

This issue is for creation of a longer form vignette. This vignette would show a fuller picture of the package's potential functionality than the getting started vignette. Here, DDD refers to documentation-driven development (#43).

## Context

The package is undergoing refactoring. As part of the refactoring we will be editing existing functions and adding new functions. To help us make these choices we need to determine which user workflows we would like to support. Once way to do this is to start to write out those user workflows. This vignette serves as one way to do this.

## Required features

- An `.Rmd` showing a longer form use of the `epidist` package
- Exact details of the use remain to be determined. Writing now, I anticipate the vignette to include:
  - Loading the Ebola data
  - Transforming the Ebola data to be compatible with the model function
  - Fitting a model to the Ebola data with a lognormal delay
  - Fitting a model to the Ebola data with a gamma delay
  - Extracting posterior distributions from the lognormal and gamma models
  - Comparing the fitted distributions with the two different models

## Out of scope

- This issue could be considered resolved when the longer form vignette is written, even if the functions are not all currently working. That is, it is not required that the vignette all runs to consider this issue complete, just that we are happy with the proposed functions and workflow.

## Related documents

- #43
- https://github.com/epinowcast/epidist/pull/32

**Comment by athowes** [comment]
Requesting feedback on 1. issue design, and 2. contents of issue @kgostic @seabbs @parksw3

**Comment by seabbs** [comment]
This looks good to me. I think we might also want to consider: 

- Visualising the delay data as an initial step
- Posterior predictions for a single or both models (may be included in your above) 
- Translating a fitted distribution to a pmf (this is most likely of my suggestions to be out of scope here)

**Comment by kgostic** [comment]
I agree with @seabbs. Structure of the issue looks great. Visualizing the delay and generating posterior predictions are both in scope and should be included.

> Translating a fitted distribution to a pmf (this is most likely of my suggestions to be out of scope here)
I would also like this to be in scope! I think it's important.

From our previous discussion, it sounded like the main blocker is that @seabbs would like the function used to convert the fitted pdf into a double censored (discrete) pmf to live in another yet-to-be-created package that doesn't require heavy stan dependencies. I'd propose just putting that function here for now or using an existing one from epinow2 or epinowcast until it can be moved?

**Comment by seabbs** [comment]
>  I'd propose just putting that function here for now or using an existing one from epinow2 or epinowcast until it can be moved?

Yes either this or just using a placeholder function name for now and seeing where we are when we get to this point

**Comment by kgostic** [comment]
@athowes, here's the relevant function from epinowcast: https://github.com/epinowcast/epinowcast/blob/e9a7a7cc9c2e02bab20491edea3bc0dc5ec2899b/R/model-module-helpers.R#L297

**Comment by athowes** [comment]
**Generic question**: after recieving feedback where I think I'd like to update the scope of the issue, is it better that I update the initial issue, or that I note that I'm updating the issue in a comment in the thread?

Adding:
* Visualising the delay data as an initial step
* Posterior predictions for a single or both models (this clarifies what I had "extracting posterior distributions" -- still unclear exactly what is being predicted here)
    
Nice to have / with pseduocode:
* Translating a fitted distribution to a PMF

**Comment by seabbs** [comment]
> fter recieving feedback where I think I'd like to update the scope of the issue

I would typically update the initial issue with a note that I have done so and a link to the comment etc that motivated it. Not sure if there is a good practice guide for this but I like it as it makes it easier to find info when coming from a PIR

**Comment by seabbs** [comment]
> Posterior predictions for a single or both mod

You can plot the fit to the actual data as eval metric. We use this @parksw3 paper for the ebola case study I believe

**Comment by parksw3** [comment]
>> Posterior predictions for a single or both models (this clarifies what I had "extracting posterior distributions" -- still unclear exactly what is being predicted here)

> You can plot the fit to the actual data as eval metric. We use this @parksw3 paper for the ebola case study I believe

I feel there are several other ways we can visualize the posterior. In our paper, we show truncated mean vs empirical delay, but other people might prefer to see pdfs and cdfs?

**Comment by athowes** [comment]
Based on discussion with @seabbs, I am coming back to this vignette as a good vehicle for completion of `epdist` 0.1.0:

* Now have functionality to fit the model by location-age-sex etc.
* Good setting to test post-processing ability of `tidybayes` etc.
* Consider this vignette as required for demonstration of package's 0.1.0 functionality
* Decided to drop the section on "retrospective analysis" and focus on features introduced
* Can write plots into this vignette, then generalise them to packaged code
* Would do this based on output from `tidybayes` etc.
* Clarify workflows required

**Comment by athowes** [comment]
# Status update

This now has models

1. `fit` i.e. `~ 1`
2. `fit_sex` i.e. `~ 1 + sex`
3. `fit_sex_district` i.e. `~ 1 + sex + district`

## Things to be included

- [ ] Figure showing the data
  - Is what I have currently sufficient?
- [ ] Call to `summary` on `fit` objects
  - Base `brms`
  - Already can be done
  - Gives an overview of lots of useful information
- [ ] Some type of plot showing posterior predictions
  - `tidybayes` with `posterior`
  - Currently blocked by #214
  - See https://mjskay.github.io/tidybayes/articles/tidy-posterior.html#posterior-predictions
  - I am interested to discuss what the actual point of this is. What research question is this answering?
  - Plot the posterior predictive distribution on top of the data somehow?
- [ ] Some type of plot showing posterior expectations
  - `tidybayes` with `posterior`
  - Currently blocked by #214 (but that part is done)
  - In my opinion the research question here is clearer: I want to know about differences by sex and district. Are they big? Are they significant?
- [ ] Translating a fitted distribution to a PMF
  - This is important because I want to use these PMF in future works
- [ ] Model comparison with `loo`
  - `loo`
  - Currently blocked by #240
  - This is important because I need to decide which model to use. Lognormal or gamma? Do the covariates do much?
